version: "3"

services:
    
  app:
    build: .
    ports:
      - 3000:3000
    volumes:
      - .:/home/node/app
    extra_hosts:
      - "host.docker.internal:host-gateway"
    depends_on:
      - mysql
    networks:
      - next_gpt

  test:
    restart: on-failure
    container_name: app_test
    build: .
    command: npm run test
    volumes:
      - .:/home/node/app
    extra_hosts:
      - "host.docker.internal:host-gateway"
    depends_on:
      - app
    networks:
        - next_gpt

  dev_test:
    container_name: app_dev_test
    build: .
    restart: on-failure
    command: npm run test:dev:staged
    volumes:
      - .:/home/node/app
    extra_hosts:
      - "host.docker.internal:host-gateway"
    depends_on:
      - test
    networks:
      - next_gpt


  mysql:
    image: mysql:8.0.30-debian
    restart: always
    environment:
      - MYSQL_ROOT_PASSWORD=root
      - MYSQL_DATABASE=chat
    volumes:
      - .docker/mysql/data:/var/lib/mysql
    ports:
      - 3307:3306
    cap_add:
      - SYS_NICE # CAP_SYS_NICE
    extra_hosts:
      - "host.docker.internal:host-gateway"
    networks:
      - next_gpt

  adminer:
    container_name: db_adminer
    image: adminer:latest
    restart: always
    environment:
      - ADMINER_DEFAULT_SERVER=mysql
    ports:
      - 8080:8080
    extra_hosts:
      - "host.docker.internal:host-gateway"
    networks:
      - next_gpt

networks:
  next_gpt:
    external: true
 

# Mac e Linux
# /etc/hosts

# 127.0.0.1 host.docker.internal

# Windows
# C:\Windows\System32\drivers\etc\hosts
